examples/src/custom_mpl_cmm_example.py

```diff
@@ -0,0 +1,114 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Example to create a custom implementation of the ESDK-MPL ICryptographicMaterialsManager class."""
```

Comment:
Could you expand on this? Unfortunately, this is the only place where we have *any* documentation on writing a custom CMM, so we should put some extra care into this.
I would suggest adding things like
* Custom implementations must implement `get_encryption_materials` and `decrypt_materials`.
* CMMs are composable; if you just want to extend the behavior of the default CMM, you can do this (demonstrated in this example). This is easiest if you just want to add a small check to the CMM methods
* You can also write your own implementation without extending a CMM if your use case calls for it, suggest looking at implementation of default CMM to get a sense of what needs to be done. This is best if you need to fundamentally change aspects of the default CMM.

Reply:
Thanks